Ticket-booking-spring-app/src/main/java/com/example/demo/rest/RestExceptionHandler.java

```diff
@@ -0,0 +1,61 @@
+package com.example.demo.rest;
+
+import com.example.demo.exception.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class RestExceptionHandler {
+    @ExceptionHandler
+    public ResponseEntity<ErrorResponse> handleException(InvalidReservationException exc) {
+        ErrorResponse error = new ErrorResponse();
```

Comment:
wszystkie te 5 metod robi dokładnie to samo, można było to wrzucić do jednej prywatnej metody z parametrami `Exception exception, int httpStatus`

Reply:
Tak, tutaj chodziło mi o to żeby w kodzie jasno było widać jakie wyjątki są obsługiwane i jaki każdy z nich daje kod błędu. Rzeczywiście wyszedł trochę powtarzalny kod :)